src/MigrationTools.Clients.AzureDevops.Rest/Processors/AzureDevOpsPipelineProcessor.cs

```diff
@@ -473,6 +475,47 @@ private void UpdateQueueIdOnPhases(ReleaseDefinition definitionToBeMigrated, IEn
             }
         }
 
+        private void UpdateServiceConnectionId(ReleaseDefinition definitionToBeMigrated, IEnumerable<Mapping> ServiceConnectionMappings)
+        {
+            if (ServiceConnectionMappings is null)
+            {
+                return;
+            }
+
+            foreach (var environment in definitionToBeMigrated.Environments)
+            {
+                foreach (var deployPhase in environment.DeployPhases)
+                {
+                    foreach (var workflowTask in deployPhase.WorkflowTasks)
+                    {
+                        bool hasFoundInputWhichNeedsReplacement = false;
+                        string inputNameWhichNeedsValueReplacement = "azureSubscription";
```

Comment:
I'm not sure whether this is the right approach to fix the issue with missing service connection mappings. Even though this might work for your specific case, it won't work for anything else.
I just tried this out and it didn't work because the service connection Id wasn't stored inside the Task but rather inside the stage parameters.

Maybe all "Update...Id" methods should be converted into a single one that goes through all properties of an object (Build Definition, Release Pipeline or Task Group) and looks out for all Ids that exist inside the mappings list.

Reply:
> I'm not sure whether this is the right approach to fix the issue with missing service connection mappings. Even though this might work for your specific case, it won't work for anything else. I just tried this out and it didn't work because the service connection Id wasn't stored inside the Task but rather inside the stage parameters.
> 
> Maybe all "Update...Id" methods should be converted into a single one that goes through all properties of an object (Build Definition, Release Pipeline or Task Group) and looks out for all Ids that exist inside the mappings list.

Well I didn't think of that during the development, but you're completely right that it wouldn't solve all cases. 

With stage parameters, do you mean that little link icon where it links tasks' input to Parameters that is listed under the stage section?

Earlier this week I did notice too that a either a BuildDefinition or ReleaseDefinition's property `ProcessParameter` was not deserialized after calling the API because the type's class (which was also called `ProcessParameter`) did not contain any properties and is defined as partial. If you query the definition via Postman you get alot of properties, of which some also need a mapping, but since the object was not entirely filled with this data, you would not be able to replace it.